src/core/lib/channel/metrics.cc

```diff
@@ -42,13 +39,12 @@ GlobalInstrumentsRegistry::RegisterUInt64Counter(
     absl::Span<const absl::string_view> optional_label_keys,
     bool enable_by_default) {
   auto& instruments = GetInstrumentList();
-  std::for_each(instruments.begin(), instruments.end(),
-                [&](const GlobalInstrumentDescriptor& descriptor) {
-                  if (descriptor.name == name) {
-                    Crash(absl::StrFormat(
-                        "Metric name %s has already been registered.", name));
-                  }
-                });
+  for (const auto& descriptor : instruments) {
```

Comment:
Done.

Reply:
Sorry, I think I steered you wrong here -- seeing this made me realize that the limitation of 64 is being imposed in the OT plugin, not here in the instrument registry, so we should enforce the limit in the OT plugin instead.

Reply:
Removed. `std::bitset<N>::set()` function will throw en exception if it is larger than what it can hold so I think we already have that there.

Reply:
Depending on an exception from `std::bitset<N>::set()` doesn't seem quite as good, because someone could define a metric with 65 optional labels, and it wouldn't trigger an exception until someone enables the 65th optional label.

Reply:
Good point. Done. Though I still like the idea to have `static_assert()` in metrics.cc to check this limit during compile-time when we have better compile-time enforcement. But I also understand that this limit is imposed by the OT plugin so it shouldn't affect the instrument registry.